packages/lumx-react/src/components/text/Text.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { forwardRef } from 'react';
+
+import { Color, ColorVariant, Typography } from '@lumx/react';
+import { Comp, GenericProps, getRootClassName, handleBasicClasses, HeadingElement } from '@lumx/react/utils';
+import classNames from 'classnames';
+
+type TextComponents = 'span' | 'p' | HeadingElement;
+
+/**
+ * Defines the props of the component.
+ */
+export interface TextProps extends GenericProps {
+    /**
+     * Color variant.
+     */
+    color?: Color;
+    /**
+     * Lightened or darkened variant of the selected color.
+     */
+    colorVariant?: ColorVariant;
+    /**
+     * Typography variant.
+     */
+    typography?: Typography;
+    /**
+     * Custom component to render the text.
+     */
+    as?: TextComponents;
+    /**
+     * Control whether the text should truncate or not.
+     * Setting as `true` will make the text truncate on a single line.
+     * Setting as `{ lines: number}` will make the text truncate on a multiple lines.
+     */
+    truncate?: boolean | { lines: number };
+}
+
+/**
+ * Component display name.
+ */
+const COMPONENT_NAME = 'Text';
+
+/**
+ * Component default class name and class prefix.
+ */
+const CLASSNAME = getRootClassName(COMPONENT_NAME);
+
+/**
+ * Component default props.
+ */
+const DEFAULT_PROPS = {
+    as: 'span',
```

Comment:
This actually was my biggest gripe about the component.
IMO the default element should be a `p` and not a `span`, for accessible reasons.

Setting it as `span` by default means we end up with components looking like this 

```jsx
<article>
  <h2>My Title</h2>
  <Text>The description of the article</Text>
</article>
```

Which will render the following html 

```html
<article>
  <h2>My Title</h2>
  <span>The description of the article</span>
</article>
```

which is neither valid nor accessible html.

Unless there are some usecases I can't think of right now, a `span` component should only be used within an element that already has semantics (headings / paragraphs), for styling purposes (or to add some other attribute).

Which means we need to add `as="p"` more times than we would need to ad `as="span"`.